### DISCUSSION TITLE
PI algorithms and objective functions

### DISCUSSION THREAD

**Discussion by PavelBal** [Open-Systems-Pharmacology/OSPSuite.ParameterIdentification#27] [Ideas]
Dear users,

the most crutial parts of a parameter identification are the identification algorithms and the objective function that calculates the distance between observed data and simulated results. In this discussion, we would like to hear your thoughts about the algorithms and objective functions that the parameter identification package should provide. Any suggestions, including those based on your experience, are highly welcomed.

You may propose an R package, an implemented algorithm, or also just the mathematical description of the objective function that we could implement.

**Comment by PavelBal** [comment]
@Yuri05 @msevestre @sfrechen

Also feel free to tag anyone who might be interested.

**Reply by rfaelens** [reply]
I think there are two aspects to consider.
First, the objective function itself. I suggest three options:
1) least squares
2) log likelihood based on a residual error model
3) log likelihood combined with a parameter prior

Second, do you want to identify a mixed effect model? Parameter identification across multiple subjects sounds worthwhile (cf Bayer presentation at Saarbrücken pkpd expert treffen). I would suggest to use mcmc for this, resulting in a sample set of likely parameter vectors.

**Comment by PavelBal** [comment]
@rfaelens Great input, thanks!
